Sources/WalletConnectRelay/RelayClient.swift

```diff
@@ -98,65 +110,69 @@ public final class RelayClient {
 
     /// Completes when networking client sends a request, error if it fails on client side
     public func publish(topic: String, payload: String, tag: Int, prompt: Bool = false) async throws {
-        let params = RelayJSONRPC.PublishParams(topic: topic, message: payload, ttl: defaultTtl, prompt: prompt, tag: tag)
-        let request = JSONRPCRequest<RelayJSONRPC.PublishParams>(method: RelayJSONRPC.Method.publish.method, params: params)
-        logger.debug("Publishing Payload on Topic: \(topic)")
-        let requestJson = try request.json()
-        try await dispatcher.send(requestJson)
+        let request = Publish(params: .init(topic: topic, message: payload, ttl: defaultTtl, prompt: prompt, tag: tag))
+            .wrapToIridium()
```

Comment:
It's easy to miss this wrap on new request implementation. We need to configure it once IG

Reply:
In one of my first experiments, the wrap + transform into RPC needed to be called only in a single place before sending the message, but I couldn't fit it in the current implementation (because of the callback problems that I was trying to solve). I agree that it should be called just once.